lib/matplotlib/widgets.py

```diff
@@ -297,10 +298,19 @@ def __init__(self, ax, label, valmin, valmax, valinit=0.5, valfmt=None,
 
         valstep : float, default: None
             If given, the slider will snap to multiples of *valstep*.
+            If both *valsnap* and *valstep* are set an error is raised.
 
         orientation : {'horizontal', 'vertical'}, default: 'horizontal'
             The orientation of the slider.
 
+        initcolor : color or None, default: 'r'
+            The color of the line drawn at the *valinit* position. If None
+            then no line marking the inital position will be drawn.
+
+        valsnap : 1D array-like, default: None
+            If given, the slider will snap to nearest value in *valsnap*.
+            If both *valsnap* and *valstep* are set an error is raised.
```

Comment:
I'm not a fan of mutually exclusive arguments. Would it be too awkward to accept array-like for *valstep*. (Steps don't need to be equal in size and "You give the step positions explicitly").

Reply:
I think it would actually simplify the implementation. I had avoided because I tend to find arguments with dual means like that to be confusing, especially if the names seems to clearly preference one interpretation. 

That said, I could've sworn I had a goto example of where I've found this confusing, but I can't actually think of an example. Plus this seems reasonably niche, and with adequate documentation it should still be very discoverable.

I will update it to only accept valstep as either a float or an arraylike.